playwright-e2e/dsm/pages/tissue-information-page/tissue-information-page.ts

```diff
@@ -0,0 +1,294 @@
+    date2 && await this.fillFaxSentDate(1, date2);
+    date3 && await this.fillFaxSentDate(2, date3);
+  }
+
+  public async fillTissueReceivedDate(date: FillDate): Promise<void> {
+    const tissueReceivedLocator = this.locatorFor(DynamicFieldsEnum.TISSUE_RECEIVED);
+    await expect(tissueReceivedLocator, 'Tissue Received Date picker is not visible').toBeVisible();
+
+    await this.fillDate(tissueReceivedLocator, date);
+  }
+
+  public async fillNotes(value: string): Promise<void> {
+    const notesLocator = this.locatorFor(DynamicFieldsEnum.NOTES);
+    await expect(notesLocator, 'Notes textarea is not visible').toBeVisible();
+
+    const textarea = new TextArea(this.page, {root: notesLocator});
+    const isTextareaDisabled = await textarea.isDisabled();
+    const existingValue = await textarea.currentValue;
+
+    if (!isTextareaDisabled && existingValue !== value) {
+      await textarea.fill(value, false);
+      await textarea.blur();
+      await waitForResponse(this.page, {uri: 'patch'});
+    }
+  }
+
+  public async fillDestructionPolicy(value: number, keptIndefinitelySelection = false, applyToAll = false): Promise<void> {
+    const destructionPolicyLocator = this.locatorFor(DynamicFieldsEnum.DESTRUCTION_POLICY);
+    await expect(destructionPolicyLocator, 'Destruction Policy Years is not visible').toBeVisible();
+
+    const destructionPolicyYears = new Input(this.page, {root: destructionPolicyLocator});
+    const isInputDisabled = await destructionPolicyYears.isDisabled();
+
+    await this.checkCheckbox(destructionPolicyLocator, keptIndefinitelySelection);
+
+    const existingValue = await destructionPolicyYears.currentValue;
+    const isAllowedToEnterValue = !keptIndefinitelySelection && !isInputDisabled && existingValue.trim() !== value.toString();
+
+    if (isAllowedToEnterValue) {
+      await destructionPolicyYears.fillSimple(value.toString());
+      await waitForResponse(this.page, {uri: 'patch'});
+    }
+
+    applyToAll && await this.applyToAll(destructionPolicyLocator);
+  }
+
+  public async problemsWithTissue(newValue: ProblemWithTissueEnum, unableToObtainSelection = false): Promise<void> {
+    const problemsWithTissueLocator = this.locatorFor(DynamicFieldsEnum.PROBLEM_WITH_TISSUE);
+    await expect(problemsWithTissueLocator, 'Problems with Tissue selection is not visible')
+      .toBeVisible();
+
+    const selectElement = new Select(this.page, {root: problemsWithTissueLocator});
+    const selectedValue = await selectElement.currentValue;
+    const isDisabled = await selectElement.isSelectDisabled();
+    const allowSelection = !isDisabled && selectedValue?.trim() !== newValue;
+
+    if (allowSelection) {
+      await selectElement.selectOption(newValue);
+      await waitForResponse(this.page, {uri: 'patch'});
+    }
+
+    await this.checkCheckbox(this.problemsWithTissueUnableToObtainCheckbox, unableToObtainSelection);
+  }
+
+  public async selectGender(gender: 'Male' | 'Female'): Promise<void> {
```

Comment:
I think we have more gender types now than just two.

Reply:
I saw there only 2